infrastructure/filefilter/filter.go

```diff
@@ -0,0 +1,215 @@
+package filefilter
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/rs/zerolog/log"
+	ignore "github.com/sabhiram/go-gitignore"
+	"gopkg.in/yaml.v3"
+)
+
+func FindNonIgnoredFiles(rootFolder string) <-chan string {
+	return newRepoIgnoresFilter(rootFolder).
+		findNonIgnoredFiles()
+}
+
+type repoIgnoresFilter struct {
+	// The path to the root of the repository
+	repoRoot               string
+	ignoreFiles            []string
+	globsPerFolder         map[string][]string
+	ignoreCheckerPerFolder map[string]ignore.IgnoreParser
+}
+
+func newRepoIgnoresFilter(rootFolder string) *repoIgnoresFilter {
+	return &repoIgnoresFilter{
+		repoRoot:               rootFolder,
+		ignoreFiles:            []string{".gitignore", ".dcignore", ".snyk"},
```

Comment:
This set should be dictated by `/filters` endpoint.

Reply:
Not really - this is not Snyk Code specific, even though it's used by Snyk Code.